astropy/nddata/nddata.py

```diff
@@ -299,5 +299,13 @@ def uncertainty(self, value):
             # If it is a subclass of NDUncertainty we must set the
             # parent_nddata attribute. (#4152)
             if isinstance(value, NDUncertainty):
+                # In case the uncertainty already has a parent create a new
```

Comment:
This chunk is complicated. When I fixed the memory leak (see #4825) it revealed a test-failure that was fixed in this chunk.